### ISSUE TITLE
NeovimSSHProvider:get_user_selection never exits

### ISSUE THREAD

**Issue by NilujePerchut** [amitds1997/remote-nvim.nvim#37]
**Describe the bug**
During `RemoteStart`, `NeovimSSHProvider:get_user_selection` always hit the `coroutine.yield()` statement and never reach the end of the `get_user_selection` function.

**To Reproduce**
I just used my configuration (https://github.com/NilujePerchut/configs/tree/master/nvim) and add the following config:
```
  {
    "amitds1997/remote-nvim.nvim",
    tag = "v0.0.1",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "MunifTanjim/nui.nvim",
      "rcarriga/nvim-notify",
      "nvim-telescope/telescope.nvim",
    }
  }
```
and
`require("remote-nvim").setup()`

**Expected behavior**
The `RemoteStart` command does not perform the remote connection

**System Info**
* Your local OS: Manjaro
* Local Neovim version: v0.9.1
* Remote host OS (if applicable): Manjaro
* Remote Neovim version (if applicable): v0.9.1

**Additional context**
If I edit a workspace by hand, the `RemoteStart` command works flawlessly.

**Comment by amitds1997** [comment]
Thanks for reporting the issue! `NeovimSSHProvider:get_user_selection` uses `vim.ui.select` which operates async so the callback should resume the coroutine once a selection is made and the coroutine remains paused until then.

Could you please provide a screen cast showing the issue you are facing? I could not gather the exact scenario that you are running into. While doing that, could you also gather the debug logs? You can follow [this comment](https://github.com/amitds1997/remote-nvim.nvim/issues/36#issuecomment-1684710438) about how to generate those logs.

**Comment by NilujePerchut** [comment]
Thank you very much for your answer (and this useful plugin !).

Here follows a screencast of the my actions.
I am connecting to osdegirouette host using an entry in my .ssh/config.

[screencast.webm](https://github.com/amitds1997/remote-nvim.nvim/assets/20670601/473c8261-1b28-49d3-ba8d-e39984e93235)

As you can see, as soon as I press enter after entering "2" to select nvim v0.9.1, `RemoteStart` get stuck.

After a few debugging I can tell that the command is stuck in the  `NeovimSSHProvider:get_user_selection` function and never reach the end of the function. I can see the callback be called. But nothing more. As I said in my first post it takes the `coroutine.yield() ` statement and never reach the end of the `get_user_selection` function.

I hope this will help.

I wanted to attach the log file but I can't find it.

Thanks in advance for your help.

**Comment by amitds1997** [comment]
> Here follows a screencast of the my actions. I am connecting to osdegirouette host using an entry in my .ssh/config.
> screencast.webm
> 
> As you can see, as soon as I press enter after entering "2" to select nvim v0.9.1, `RemoteStart` get stuck.
> 
> After a few debugging I can tell that the command is stuck in the `NeovimSSHProvider:get_user_selection` function and never reach the end of the function. I can see the callback be called. But nothing more. As I said in my first post it takes the `coroutine.yield() ` statement and never reach the end of the `get_user_selection` function.

Thanks for the screencast. The issue was indeed in the `get_user_selection` method. The default implemenatation of `vim.ui.select` is synchronous (it calls `:inputlist`). The previous implementation assumed that the implementation would be asynchronous (something that is true with plugins like `dressing.nvim` which override the interface).

The issue should be fixed by [fb2c301](https://github.com/amitds1997/remote-nvim.nvim/pull/38/commits/fb2c3011d5fdbc114b1d87cac87e764c1b8f5158) and you can check out the `master` branch to test it out. Do let me know if you still see this issue.

**Comment by NilujePerchut** [comment]
It's working great now.
Thank you very much !
